### PR TITLE
serial: Replace the serial-rs crate with serialport-rs

### LINKED ISSUE #86: serial-unix crate is unmaintained
We have noticed that this crate uses serial-unix which has not been maintained in 5 years: https://github.com/dcuddeback/serial-rs. 

Instead the maintained crate serialport (https://github.com/serialport/serialport-rs) could be used.

### PR DESCRIPTION
**Pull request by qrasmont**
Replace the unmaintained serial-rs crate with serialport-rs.
Closes #86

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
Thank you!
Could you add the libudev-dev package [here](https://github.com/rust-embedded/linux-embedded-hal/blob/0a65cd15e3a9fc78162ac77342abfa5613059737/.github/workflows/ci.yml#L44) and also add an entry to the changelog?

**Comment by qrasmont** [comment]
> Thank you! Could you add the libudev-dev package [here](https://github.com/rust-embedded/linux-embedded-hal/blob/0a65cd15e3a9fc78162ac77342abfa5613059737/.github/workflows/ci.yml#L44) and also add an entry to the changelog?

I chose to remove the dependency to libudev-sys (by disabling default features for the serialport-rs crate) because cross compilation would rely on a pkg-config config with a sysroot for the target (see [readme](https://github.com/dcuddeback/libudev-sys#cross-compiling)) which complicates the setup for CI and usage.
Especially since as described [here](https://github.com/serialport/serialport-rs#overview), the impact of disabling this dependency  sums up as : "Enumeration will still be available [...], but won't expose as much information and may return ports that don't exist physically."

This would not be necessary if this PR got merged into libudev-sys: https://github.com/dcuddeback/libudev-sys/pull/8

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- [breaking-change] Replace serial-rs with the serialport-rs crate.
```
Comment (eldruin) [inline CHANGELOG.md:10]:
```suggestion
### Changed
- [breaking-change] Replace serial-rs with the serialport-rs crate. `Serial::open` now needs a baud-rate argument as well.
```

**Review by eldruin — COMMENTED** [review]
Great solution. Thank you for looking into it.
Sadly libudev-sys seems abandoned, but great to know that it is not really necessary.

**Review by eldruin — APPROVED** [review]
Looks good to me, thank you for your work!
bors r+

**Pull request merged**